src/plugins/kibana/public/visualize/editor/agg_group.js

```diff
@@ -40,6 +42,29 @@ uiModules
           if (count < schema.max) return true;
         });
       });
+
+      const drake = dragula({
+        containers: $('.vis-editor-agg-group', $el).toArray(),
+        moves(el, source, handle) {
+          const $handle = $(handle);
+          return $handle.add($handle.parent()).is('.vis-editor-agg-header-drag-handle');
+        }
+      });
+
+      drake.on('drop', (el, target, source, sibling) => {
+        const aggs = $scope.group;
+        const aggScope = $(el).scope();
+        _.move(aggs, aggScope.agg, getIndex(sibling));
+        function getIndex(el) {
+          if (!el) { return aggs.length - 1; }
+          const list = $('.vis-editor-agg-wrapper', target);
+          return list.index(el) - 1;
```

Comment:
Since we have access to the agg, and the sibling, can we do the re-ordering using the position of aggs in `$scope.group` rather than using the position of dom elements?